desci-server/src/controllers/nodes/publish.ts

```diff
@@ -232,107 +457,17 @@ export const publishHandler = async ({
-       * There can be multiple mirrors per node, right now there is just Estuary
-       *
-       * NOTE: uncomment when reactivating public ref mirroring
-       const dataMirrorJobs = await createDataMirrorJobs(cidsRequiredForPublish, owner.id);
-       */
-
-      // TODO: update public data refs to link versionId
-
-      /**
-       * Save a success for configurable service quality tracking purposes
-       */
-      await saveInteractionWithoutReq(ActionType.PUBLISH_NODE_CID_SUCCESS, {
-        cidsPayload,
-        result: { newPublicDataRefs },
-      });
-    } catch (error) {
-      logger.error({ error }, `[publish::publish] error=${error}`);
-      /**
-       * Save a failure for configurable service quality tracking purposes
-       */
-      await saveInteractionWithoutReq(ActionType.PUBLISH_NODE_CID_FAIL, { cidsPayload, error });
-      throw error;
-    }
+    await handlePublicDataRefs({
+        nodeId: node.id,
+        nodeUuid: node.uuid,
+        userId: owner.id,
+        manifestCid: cid,
+        nodeVersionId: nodeVersion.id,
+    });
 
-    /**
-     * Publish Step 2:
-     * Initiate IPFS storage upload using Estuary
-     */
     const manifest = await getManifestByCid(cid);
-
     const targetDpidUrl = getTargetDpidUrl();
-
-    // const researchObjectToPublish = { uuid, cid, manifest, ownerId: owner.id };
-    const sendDiscordNotification = (error: boolean) => {
-      const manifestSource = manifest as ResearchObjectV1;
-      discordNotify(`${targetDpidUrl}/${manifestSource.dpid?.id}${error ? ' (note: estuary-err)' : ''}`);
-    };
-
-    // Send an email update to all contributors
-    // await publishServices.sendVersionUpdateEmailToAllContributors({ node });
-
-    /**
-     * NOTE: uncomment when reactivating public ref mirroring
-    const handleMirrorSuccess = async (publishedResearchObjectResult) => {
-      await saveInteractionWithoutReq(ActionType.PUBLISH_NODE_RESEARCH_OBJECT_SUCCESS, {
-        researchObjectToPublish,
-        result: publishedResearchObjectResult,
-      });
-
-      sendDiscordNotification(false);
-    };
-    const handleMirrorFail = async (error) => {
-      await saveInteractionWithoutReq(ActionType.PUBLISH_NODE_RESEARCH_OBJECT_FAIL, {
-        researchObjectToPublish,
-        error,
-      });
-
-      sendDiscordNotification(true);
-    };
-
-    const publicDataReferences = await prisma.publicDataReference.findMany({
-      where: {
-        versionId: nodeVersion.id,
-      },
-    });
-    logger.debug(
-      { publicDataReferences },
-      `[publish::publish] publicDataReferences=${JSON.stringify(publicDataReferences)}`,
-    );
-
-    // trigger ipfs storage upload, but don't wait for it to finish, will happen async
-    publishResearchObject(publicDataReferences).then(handleMirrorSuccess).catch(handleMirrorFail);
-    */
-    sendDiscordNotification(false);
+    discordNotify(`${targetDpidUrl}/${manifest.dpid?.id}`);
```

Comment:
why are there 2 paths to discordNotify?

Reply:
Mutually exclusive: one is in the old async publish handle, the other is in `syncPublish`. The potentially new dPID necessary to build the link isn't available in the request receiver :)